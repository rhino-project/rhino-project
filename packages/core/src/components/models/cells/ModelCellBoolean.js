import { useGlobalOverrides } from 'rhino/hooks/overrides';
import CellBoolean from 'rhino/components/table/cells/CellBoolean';

export const ModelCellBooleanBase = (props) => <CellBoolean {...props} />;

const defaultComponents = { ModelCellBoolean: ModelCellBooleanBase };

const ModelCellBoolean = ({ overrides, ...props }) => {
  const { ModelCellBoolean } = useGlobalOverrides(
    defaultComponents,
    overrides,
    props
  );

  return <ModelCellBoolean {...props} />;
};

export default ModelCellBoolean;
