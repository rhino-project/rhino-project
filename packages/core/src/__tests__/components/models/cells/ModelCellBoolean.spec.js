import ModelCellBoolean from 'rhino/components/models/cells/ModelCellBoolean';
import { sharedCellTests } from './sharedCellTests';

describe('ModelCellBooelan', () => {
  sharedCellTests(ModelCellBoolean);
});
