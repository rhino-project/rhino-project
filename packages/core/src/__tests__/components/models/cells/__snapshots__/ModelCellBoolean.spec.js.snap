// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelCellBooelan should render the empty text when value is nullish 1`] = `
<DocumentFragment>
  -
</DocumentFragment>
`;

exports[`ModelCellBooelan should render the overridden empty text when value is nullish 1`] = `
<DocumentFragment>
  baz
</DocumentFragment>
`;

exports[`ModelCellBooelan should render with global override shorthand 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;

exports[`ModelCellBooelan should render with global override shorthand for model 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;

exports[`ModelCellBooelan should render with global override shorthand for model and attribute 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;
